test/promql/tests/data_plane/component_errors_test.yaml

```diff
@@ -0,0 +1,89 @@
+evaluation_interval: 1m
+
+rule_files:
+  - prometheus.component_alerts.yaml
+
+tests:
+  # ----- Component Deletion Alerting Tests -----
+  - interval: 1m
+    input_series:
+
+      # HAS is experiencing component deletion failure rate of 100%, so it will be alerted
+      - series: 'has_component_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x64'
+      - series: 'has_component_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '1+1x64'
+
+    alert_rule_test:
+      - eval_time: 65m
+        alertname: ComponentDeletionErrors
+        exp_alerts:
+          - exp_labels:
+              severity: warning
+              slo: true
+              namespace: application-service
+              pod: has
+              source_cluster: cluster01
+            exp_annotations:
+              summary: >-
+                HAS is experiencing component deletion failures of >5%
+              description: >-
+                Component controller in Pod has for namespace
+                application-service on cluster cluster01 is failing to
+                successfully delete at least 95% of components over the past hour
+              runbook_url: https://gitlab.cee.redhat.com/rhtap/docs/sop/-/blob/main/has/component-create-failed.md
+
+  - interval: 1m
+    input_series:
+
+      # HAS experienced ~10% (10.9%) deletion failure rate, should still alert
+      - series: 'has_component_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '10+10x64'
+      - series: 'has_component_failed_deletion_total{namespace="application-service", pod="has", source_cluster="cluster01"}'
+        values: '0x20 0+10x7 70x38'
```

Comment:
The counter is resetting here. It's a good test case.

I think we need one in which the counter does not reset as well.
Also, I think including the query value on the output will make it clear if the reset is being taken into account.

Reply:
The counter is not really resetting.  The scenario represents no failures for first 20 mins, then an incrementing series of failures for the next 7 mins until it plateaus.    I'll try to figure out a counter reset scenario

Reply:
You're right. I got confused.